src/compose/_tests_/placeholder-test.js

```diff
@@ -0,0 +1,58 @@
+import getInputPlaceholder from '../getInputPlaceholder';
+import {
+  privateNarrow,
+  streamNarrow,
+  topicNarrow,
+  groupNarrow,
+} from '../../utils/narrow';
+
+describe('getInputPlaceholder returns a smart placeholder for message compose text box. It depends on narrow', () => {
+  test('returns "Message @ThisPerson" for person narrow', () => {
+    const narrow = privateNarrow('abc@zulip.com');
+
+    const auth = { realm: 'http://localhost:9991',
+      email: 'hamlet@zulip.com',
+      apiKey: 'apiKey' };
+
+    const users = [
+      { id: 23,
+        email: 'abc@zulip.com',
+        fullName: 'ABC',
+      },
+      { id: 22,
+        email: 'xyz@zulip.com',
+        fullName: 'XYZ',
+      },
+    ];
+
+    expect(getInputPlaceholder(narrow, auth, users)).toEqual('Message @ABC');
+  });
+
+  test('returns "Jot down something" for self narrow', () => {
+    const narrow = privateNarrow('abc@zulip.com');
+
+    const auth = { realm: 'http://localhost:9991',
+      email: 'abc@zulip.com',
+      apiKey: 'apiKey' };
+
+    expect(getInputPlaceholder(narrow, auth)).toEqual('Jot down something');
```

Comment:
Separate the test code by the 'Arrange, Act, Assert' principle:
http://wiki.c2.com/?ArrangeActAssert

The difference is subtle but noticeable. In that case the `getInputPlaceholder` call will be separate line. (have a look at other tests in the project)